File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/utils/RobotDrawable.java
package cz.muni.fi.umimecesky.utils;

import android.app.Activity;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;

import java.lang.reflect.Field;
import java.security.SecureRandom;
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

import cz.muni.fi.umimecesky.R;

public class RobotDrawable {

    private List<String> robotNames = new ArrayList<>();
    private Activity activity;

    private final Random RANDOM = new SecureRandom();

    public RobotDrawable(Activity activity) {

        this.activity = activity;
        Field[] drawables = R.drawable.class.getFields();
        for (Field f : drawables) {
            String fieldName = f.getName();
            if (fieldName.startsWith("robot")) {
                robotNames.add(f.getName());
            }
        }
    }

    public Drawable removeRobotDrawable() {

        Resources resources = activity.getResources();
        int randomInt = RANDOM.nextInt(robotNames.size());
        String randomDrawableName = robotNames.remove(randomInt);
        final int resourceId = resources.getIdentifier(randomDrawableName, "drawable",
                activity.getPackageName());
        return resources.getDrawable(resourceId);
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/utils/RobotDrawable.kt
package cz.muni.fi.umimecesky.utils

import android.app.Activity
import android.graphics.drawable.Drawable
import cz.muni.fi.umimecesky.R
import java.security.SecureRandom
import java.util.*

class RobotDrawable(private val activity: Activity) {

    private val robotNames = ArrayList<String>()

    private val random = SecureRandom()

    init {
        val drawables = R.drawable::class.java.fields
        for (f in drawables) {
            val fieldName = f.name
            if (fieldName.startsWith("robot")) {
                robotNames.add(f.name)
            }
        }
    }

    fun removeRobotDrawable(): Drawable {

        val resources = activity.resources
        val randomInt = random.nextInt(robotNames.size)
        val randomDrawableName = robotNames.removeAt(randomInt)
        val resourceId = resources.getIdentifier(randomDrawableName, "drawable",
                activity.packageName)
        return resources.getDrawable(resourceId)
    }
}
